### PR TITLE
Update code snippets on group pages + add copy functionality

### PR DESCRIPTION
**Pull request by jenpaulhus**
We implement code snippets for matrix groups over Fq (and over Fp for GAP).

For example:
12240.b now has a code snippet in Magma and GAP for the matrix group construction for this group.
Same for 58806.h

12.4 now has a code snippet in GAP for matrix group / F3

Also, thanks to @edgarcosta we add a copy button at the beginning of all the snippets, which could be implemented on other pages as well.

### CODE REVIEW

**Comment by jenpaulhus** [comment]
Forgot to add, this also changes how we display elements of Fq.  They are now written as alpha^{power}  where alpha is a generator of the multiplicative group.  Once this is merged, we do need to change the knowl for matrix groups to reflect this.

**Comment by jenpaulhus** [comment]
I'm not sure about this lint error.  The semicolons are there on purpose to get these particular sage commands to work.

File: lmfdb/groups/abstract/web_groups.py [hunk 1]
```diff
@@ -1922,23 +1937,47 @@ def decode_as_matrix(self, code, rep_type, as_str=False, LieType=False, ListForm
             d = rep_data["d"]
         else:
             R, N, k, d, rep_type = self._matrix_coefficient_data(rep_type)
+            if rep_type == "GLFq":
+                q=N**k
+                R = GF(q, modulus = "primitive", names=('a',)); (a,) = R._first_ngens(1) #need a for powers
```
Comment (edgarcosta) [inline lmfdb/groups/abstract/web_groups.py:1942]:
```suggestion
                R = GF(q, modulus = "primitive", names=('a',))
                (a,) = R._first_ngens(1) #need a for powers
```

File: lmfdb/groups/abstract/web_groups.py [hunk 2]
```diff
@@ -1903,7 +1915,10 @@ def _matrix_coefficient_data(self, rep_type, as_str=False):
             q = ZZ(rep_data["q"])
             if sq_flag:
                 q = q**2
-            R = rf"\F_{{{q}}}" if as_str else GF(q)
+            if as_str:
+                R = rf"\F_{{{q}}}"
+            else:
+                R = GF(q, modulus = "primitive", names=('a',)); (a,) = R._first_ngens(1)
```
Comment (edgarcosta) [inline lmfdb/groups/abstract/web_groups.py:1921]:
```suggestion
                R = GF(q, modulus = "primitive", names=('a',))
                (a,) = R._first_ngens(1)
```

**Review by edgarcosta — CHANGES_REQUESTED** [review]

File: lmfdb/groups/abstract/web_groups.py (hunk 2 above)
Comment (edgarcosta) [reply lmfdb/groups/abstract/web_groups.py:1921]: I think this is all you need to do

**Review by edgarcosta — APPROVED** [review]

**Pull request merged**